lib/src/decorators/widget_decorators.dart

```diff
@@ -10,6 +10,45 @@ import '../core/decorator.dart';
 import '../factory/mix_provider_data.dart';
 import '../helpers/lerp_helpers.dart';
 
+class IntrinsicHeightDecorator
+    extends WidgetDecorator<IntrinsicHeightDecorator> {
+  const IntrinsicHeightDecorator({super.key});
+
+  @override
+  IntrinsicHeightDecorator lerp(
+    covariant Decorator<Decorator>? other,
+    double t,
+  ) {
+    return this;
+  }
+
+  @override
+  get props => [];
+
+  @override
+  Widget build(MixData mix, Widget child) =>
+      IntrinsicHeight(key: key, child: child);
+}
+
+class IntrinsicWidthDecorator extends WidgetDecorator<IntrinsicWidthDecorator> {
+  const IntrinsicWidthDecorator({super.key});
+
+  @override
+  IntrinsicWidthDecorator lerp(
+    covariant Decorator<Decorator>? other,
```

Comment:
change type to `InstrinsicWidthDecorator? other`, also it should probably return `other` just for consistency if it exists